lib/components/viewers/stop-time-cell.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * Renders an icon based on the given stop time realtime state.
+ */
+function renderIcon (stopTime) {
+  const iconType = stopTime.realtimeState === 'UPDATED' ? 'rss' : 'clock-o'
+  return (
+    <div style={{ float: 'left' }}>
+      <Icon
+        type={iconType}
+        style={{ color: '#888', fontSize: '0.8em', marginRight: 2 }}
+      />
+    </div>
+  )
+}
+
+/**
+ * If the browser timezone is not the homeTimezone, appends a time zone designation (e.g., PDT)
+ * to designate times in the homeTimezone
+ * (e.g., user in New York, but viewing a trip planner for service based in Los Angeles).
+ */
+function getTimeFormat (timeFormat, homeTimezone) {
+  const userTimeZone = getUserTimezone()
+  const inHomeTimezone = homeTimezone && homeTimezone === userTimeZone
+  return inHomeTimezone ? timeFormat : `${timeFormat} z`
+}
+
+/**
+ * Renders a stop time as either schedule or countdown, with an optional status icon.
+ * Stop time that apply to a different day have an additional text showing the day of departure.
+ */
+const StopTimeCell = ({
+  homeTimezone,
+  showIcon,
+  soonText,
+  stopTime,
+  timeFormat,
+  useCountdown,
+  useSchedule
+}) => {
+  const departureTime = useSchedule
+    ? stopTime.scheduledDeparture
+    : stopTime.realtimeDeparture
+  const now = moment().tz(homeTimezone)
+  const serviceDay = moment(stopTime.serviceDay * 1000).tz(homeTimezone)
+
+  // Determine if arrival occurs on different day, making sure to account for
+  // any extra days added to the service day if it arrives after midnight. Note:
+  // this can handle the rare (and non-existent?) case where an arrival occurs
+  // 48:00 hours (or more) from the start of the service day.
+  const departureTimeRemainder = departureTime % ONE_DAY_IN_SECONDS
+  const daysAfterServiceDay = (departureTime - departureTimeRemainder) / ONE_DAY_IN_SECONDS
+  const departureDay = serviceDay.add(daysAfterServiceDay, 'day')
+  const vehicleDepartsToday = now.dayOfYear() === departureDay.dayOfYear()
+
+  // Determine whether to show departure as countdown (e.g. "5 min") or as HH:mm
+  // time.
+  const secondsUntilDeparture = getSecondsUntilDeparture(stopTime, useSchedule)
+  // Determine if vehicle arrives after midnight in order to advance the day of
+  // the week when showing arrival time/day.
+  const departsInFuture = secondsUntilDeparture > 0
+  // Show the exact time if the departure happens within an hour.
+  const showCountdown = useCountdown && secondsUntilDeparture < ONE_HOUR_IN_SECONDS && departsInFuture
+
+  // Use "soon text" (e.g., Due) if vehicle is approaching.
+  const countdownString = secondsUntilDeparture < 60
+    ? soonText
+    : formatDuration(secondsUntilDeparture)
+  const formattedTime = formatSecondsAfterMidnight(
+    departureTime,
+    // Only show timezone (e.g., PDT) if user is not in home time zone (e.g., user
+    // in New York, but viewing a trip planner for service based in Los Angeles).
+    getTimeFormat(timeFormat, homeTimezone)
```

Comment:
This doesn't appear to be working for me with the TriMet config `homeTimezone: America/Los_Angeles`.

Reply:
This may be related to https://momentjs.com/docs/#/displaying/format/.

Reply:
Fixed (temporarily) in https://github.com/opentripplanner/otp-react-redux/pull/292/commits/8ebacce5ee93b982c345db26b44396c6ce1329d6.

![image](https://user-images.githubusercontent.com/56846598/102843633-8f87bf80-43d7-11eb-87cd-e69f036b8100.png)

Reply:
I have added https://github.com/opentripplanner/otp-ui/issues/230 should we decide to update the library.

Reply:
Do we want to do anything regarding the layout on the pattern mode now that it shows the timezone? 
![image](https://user-images.githubusercontent.com/56846598/102901413-93e5c400-443b-11eb-890d-96e9fff89972.png)

Reply:
Hmm... I didn't think of that. Perhaps, in the interest of removing redundant information, we should change the behavior to show an alert at the top of the stop viewer (just below the "Plan a trip" line) that states that times are in a different timezone (mention the timezone and time diff).

Reply:
I like this approach too.

Reply:
Updated in https://github.com/opentripplanner/otp-react-redux/pull/292/commits/daeb6ac3811946167078ea91d3cb39f028c8fe2f.